plotly/tests/test_core/test_tools/test_TraceFactory.py

```diff
@@ -0,0 +1,112 @@
+from unittest import TestCase
+from plotly.graph_objs import graph_objs, Scatter, Data, Marker, Line, Trace
+from plotly.exceptions import PlotlyError
+
+import plotly.plotly as py
+import plotly.tools as tls
+import math
+from nose.tools import raises
+
+
+@raises(PlotlyError)
+def test_unequal_xy_length():
+    data = tls.TraceFactory.create_quiver(x=[1, 2], y=[1], u=[1, 2], v=[1, 2])
+
+
+@raises(ValueError)
+def test_wrong_scale():
+    data = tls.TraceFactory.create_quiver(x=[1], y=[1], u=[1], v=[1],
+                                          scale=0)
+
+
+@raises(ValueError)
+def test_wrong_arrow_scale():
+    data = tls.TraceFactory.create_quiver(x=[1], y=[1], u=[1], v=[1],
+                                          arrow_scale=-1)
+
+
+class TestQuiver(TestCase):
+
+    def test_one_arrow(self):
+        self.assertAlmostEqual(tls.TraceFactory.create_quiver(x=[1], y=[1],
```

Comment:
:cow2: This seems like more work that it's worth, but trust me : D

*Try and make tests as clear as possible* this is one of the places that we'll look at with a lot of scrutiny for code review and this is what keeps stuff working as expected! It's trivial to you know, but this is super helpful to future debuggers (which could be you!).

How about:

```
    def test_one_arrow(self):
    
        # we should be able to create a single arrow using create_quiver
    
        quiver = tls.TraceFactory.create_quiver(x=[1], y=[1], u=[1], v=[1], scale=1)
        expected_quiver = {
            'y': [1, 2, None, 1.615486170766527, 2, 1.820698256761928, None],
            'x': [1, 2, None, 1.820698256761928, 2, 1.615486170766527, None],
            'type': 'scatter',
            'mode': 'lines'
        }
        self.assertEqual(quiver, expected_quiver)
```